src/System.IO.Pipelines/Pipe.cs

```diff
@@ -493,29 +478,24 @@ ReadableBufferAwaitable IPipeReader.ReadAsync()
 
         private void Dispose()
         {
-            // TODO: Review throw if not completed?
```

Comment:
For "Dispose()" to be called both the reader and writer would need to be completed. Both reading and writing operations already throw if called after their respective sides complete.